### PR TITLE
releng: Update references to CIP repo to sigs.k8s.io/promo-tools

### PR DESCRIPTION
**Pull request by justaugustus**
(We'll defer alpha-sorts and file renames to a follow-up to avoid a painful review/becoming a rebase magnet.)

/hold for consensus timeout: https://groups.google.com/g/kubernetes-dev/c/YnIYtEBURvk
ref: https://github.com/kubernetes-sigs/k8s-container-image-promoter/issues/424

Signed-off-by: Stephen Augustus <foo@auggie.dev>

cc: @kubernetes/release-engineering

### CODE REVIEW

**Comment by justaugustus** [comment]
/assign @spiffxp @mrbobbytables @cblecker 
As @kubernetes/owners (though I've found there are no disruptive tasks around branch renames)

(Will fix the [labels.md](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/23605/pull-test-infra-bazel/1437940923609649152) in the next push.)

**Comment by justaugustus** [comment]
/assign @puerco @cpanato @Verolop @ameukam 
cc: @kubernetes/release-engineering @kubernetes/wg-k8s-infra

**Review by cpanato — APPROVED** [review]
/lgtm

thanks! <3

File: config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml [hunk 1]
```diff
@@ -88,11 +88,11 @@ presubmits:
         - files
         - --manifests=/home/prow/go/src/github.com/kubernetes/k8s.io/artifacts/
         - --dry-run=true
-  kubernetes-sigs/k8s-container-image-promoter:
+  kubernetes-sigs/promo-tools:
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml:91]: Rename this file/path?

Comment (justaugustus) [reply config/jobs/kubernetes/sig-release/cip/container-image-promoter.yaml:91]: @spiffxp -- Done! Rebased and renamed to `config/jobs/kubernetes/sig-release/promo-tools/promo-tools.yaml`.

File: config/jobs/kubernetes/sig-release/promo-tools/promo-tools.yaml [hunk 2]
```diff
@@ -88,11 +88,11 @@ presubmits:
         - files
         - --manifests=/home/prow/go/src/github.com/kubernetes/k8s.io/artifacts/
         - --dry-run=true
-  kubernetes-sigs/k8s-container-image-promoter:
+  kubernetes-sigs/promo-tools:
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/promo-tools/promo-tools.yaml:91]: For followup: I would prefer to see this live in config/jobs/kubernetes-sigs/promo-tools

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by justaugustus** [comment]
/hold cancel
Both this and https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/428 have been approved.

File: config/jobs/kubernetes/sig-release/promo-tools/promo-tools.yaml (hunk 2 above)
Comment (justaugustus) [reply config/jobs/kubernetes/sig-release/promo-tools/promo-tools.yaml:91]: > For followup: I would prefer to see this live in config/jobs/kubernetes-sigs/promo-tools

Addressed in https://github.com/kubernetes/test-infra/pull/23626.

**Pull request merged**